smartsim/error/errors.py

```diff
@@ -116,6 +116,10 @@ class LauncherNotFoundError(LauncherError):
     """A requested launcher could not be found"""
 
 
+class UnrecognizedLaunchedJobError(LauncherError):
```

Comment:
The other error classes start with ``Launcher``.  Would ``LauncherJobNotFound`` be better / more consistent?

Reply:
Somehow I totally missed that pattern! I'll rename to conform!